org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/resource/ISonarLintProject.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2017 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.core.resource;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import javax.annotation.CheckForNull;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.preferences.IScopeContext;
+import org.sonarlint.eclipse.core.internal.resources.SonarLintProjectConfiguration;
+
+/**
+ * Represents a project for SonarLint. A project will contain the 
+ * project level specific SonarLint configuration (binding, exclusions, ...).
+ * @since 2.7
+ */
+public interface ISonarLintProject extends ISonarLintIssuable {
+
+  /**
+   * Is the project open. Most actions are disabled on closed projects.
+   */
+  boolean isOpen();
+
+  /**
+   * Is this project already bound to a remote SonarQube project/module?
+   */
+  default boolean isBound() {
+    return SonarLintProjectConfiguration.read(getScopeContext()).isBound();
+  }
+
+  /**
+   * Is auto-analysis enabled on this project (ie on save / on open)
+   */
+  default boolean isAutoEnabled() {
+    return SonarLintProjectConfiguration.read(getScopeContext()).isAutoEnabled();
+  }
+
+  /**
+   * The scope context used to store SonarLint configuration
+   */
+  IScopeContext getScopeContext();
+
+  /**
+   * Unique name of the project that is displayed in logs and UI
+   */
+  @Override
+  String getName();
+
+  /**
+   * Physical directory that is used to compute relative path for files to be analyzed.
+   * TODO we should get rid of this at some point.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)